### PR TITLE
Improve resource deletion with node

### PR DESCRIPTION
**Pull request by binoue**
Topolvm needs a manual operation if node resource is deleted or executing `kubectl drain` as shown in #41.

To fix this, this pull request makes the following changes:

- `topolvm-metrics` now adds a finalizer to Node in addition to a metrics annotation.
- `topolvm-hook` now adds a finalizer to TopoLVM PVC.
- Add a new controller` topolvm-controller` to:
    - delete PVCs before deleting nodes, and
    - clean up stale LogicalVolume resources.
- Change LogicalVolume CRD from namespaced to cluster resource.

**This pull request contains a breaking change** because of changes to LogicalVolume CRD.
Existing installations need to be uninstalled before upgrading.

### CODE REVIEW

File: controller/controllers/persistentvolumeclaim_controller.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+		if fin == topolvm.PVCFinalizer {
+			needFinalize = true
+			break
+		}
+	}
+	if !needFinalize {
+		return ctrl.Result{}, nil
+	}
+
+	if len(pvc.Finalizers) != 1 {
```
[the remark this answers is not in the document]
Comment (masa213f) [reply controller/controllers/persistentvolumeclaim_controller.go]: Deleting running pods here means to disable `in-use-protection` forcibly even the Node is not in deleting.

On the other hand, if a Node is in deleting or drain status, Pod will be deleted by the default controller.

So here, I think we have not to delete running Pods.

[the remark this answers is not in the document]
Comment (ymmt2005) [reply controller/controllers/persistentvolumeclaim_controller.go]: Agreed.

File: hook/mutate_persistentvolumeclaim.go [hunk 2]
```diff
@@ -35,18 +32,18 @@ func (m persistentVolumeClaimMutator) Handle(ctx context.Context, req admission.
...
 	// StorageClassName can be nil
 	if pvc.Spec.StorageClassName == nil {
 		return admission.Allowed("no request for TopoLVM")
 	}
 
-	if !targets[*pvc.Spec.StorageClassName] {
+	var sc storagev1.StorageClass
+	err = m.client.Get(ctx, types.NamespacedName{Name: *pvc.Spec.StorageClassName}, &sc)
+	if err != nil {
+		return admission.Allowed("cannot get StorageClass")
```
Comment (ymmt2005) [inline hook/mutate_persistentvolumeclaim.go]: Errored?

File: hook/mutate_persistentvolumeclaim.go [hunk 3]
```diff
@@ -40,7 +40,7 @@ func (m persistentVolumeClaimMutator) Handle(ctx context.Context, req admission.
 	var sc storagev1.StorageClass
 	err = m.client.Get(ctx, types.NamespacedName{Name: *pvc.Spec.StorageClassName}, &sc)
 	if err != nil {
-		return admission.Allowed("cannot get StorageClass")
+		return admission.Errored(http.StatusBadRequest, err)
```
Comment (ymmt2005) [inline hook/mutate_persistentvolumeclaim.go]: `BadRequest` is not appropriate here.  Maybe `InternalServerError` is?

**Pull request merged**